packages/inventory-compliance/src/ComplianceRemediationButton.js

```diff
@@ -29,9 +25,13 @@ class ComplianceRemediationButton extends React.Component {
         });
     }
 
+    ruleProfile = (rule, system) => (
+        system.profiles.find(profile => profile.rules.find(profileRule => rule.refId === profileRule.refId))
+    )
+
     rulesWithRemediations = (rules, system) => {
         return rules.filter(rule => rule.remediationAvailable).map(
-            rule => this.formatRule(rule, system.profiles[0].refId, system.id)
+            rule => this.formatRule(rule, this.ruleProfile(rule, system).refId, system.id)
```

Comment:
:warning: Here's one of the keys of the PR. In this commit (https://github.com/RedHatInsights/frontend-components/commit/a5b43f91adb8dbce0c149b02f369425a526d6f9a#diff-2339b617769a4b932bc7ed3b63b7e41dR34) , I changed the old code, which was *always* returning undefined, to a code that aimed to find the profile. However, it was not accurate - as profiles[0] might not be the right one. We should find it in the list of profiles instead